core/src/main/java/org/axonframework/eventsourcing/eventstore/RevisionBasedSnapshotJury.java

```diff
@@ -0,0 +1,44 @@
+package org.axonframework.eventsourcing.eventstore;
+
+import org.axonframework.serialization.RevisionResolver;
+import org.axonframework.serialization.SerializedType;
+
+import static org.axonframework.common.ObjectUtils.getOrDefault;
+
+/**
+ * A simple {@link org.axonframework.serialization.RevisionResolver} based implementation of
+ * SnapshotJury. Decides whether to use a snapshot based on the current aggregate revision and the current snapshot
+ * revision. It simply does an 'equals' match and does not consider the order of revisions.
+ *
+ * @author Shyam Sankaran
+ * @since 3.3
+ */
+public class RevisionBasedSnapshotJury implements SnapshotJury {
+
+    private static final String NO_REVISION = "NO_REVISION";
+    private final RevisionResolver resolver;
+
+    /**
+     * Initializes a RevisionBasedSnapshotJury with given {@code resolver}.
+     *
+     * @param resolver Resolver of type {@link org.axonframework.serialization.RevisionResolver } to resolve the
+     *                 revision of aggregate corresponding to the snapshot
+     */
+    public RevisionBasedSnapshotJury(RevisionResolver resolver) {
+        this.resolver = resolver;
+    }
+
+    @Override
+    public boolean decide(DomainEventData<?> snapshot) {
+        final SerializedType payloadType = snapshot.getPayload().getType();
+        final String payloadRevision = getOrDefault(payloadType.getRevision(), NO_REVISION);
```

Comment:
I'm not happy with usage of 'magic' string here "NO_REVISION". In this case I would rather go with `null` representing that revision is missing.

Reply:
"Regarding the name of SnapshotJury, I'm not completely happy with it, but cannot think of anything better though" 
C'mon, you should be happy that I'm not charging you any legal fees :)

As to "NO_REVISION" - Yeah, I know its not the best approach. But, I did that so that I could do a equals check. Otherwise I'd have to do null checks for snapshot revision and aggregate revision separately and then also do an equals check, which will result in 2 or 3 more lines of more defensive code. I want to avoid that. Perhaps we should make AnnotationResolver and payload.getType() return an Optional in future.

Reply:
Yup, I understand why you did it, but since it is a free string that someone could supply e.g. in `@Revision` annotation, I don't think we should leave the solution as is.

Reply:
yeah, this was my thought earlier that it forces the developer to be more responsible and remember to update the version every time the aggregate is changed.

The alternate would be for the revision resolver to do some kind of class analysis and produce a revision itself. This would also mean changing the way in which snapshot is saved.

Another thing to consider is using the same resolver that was used to resolve the revision of snapshot, to resolve the aggregate, which means persisting the resolver type in addition to the payload revision.

Maybe that's where we want to end up with. But for now, I think it adds a lot of complexity and the revision based solution is good enough fix for the problem.

Reply:
I think that we are talking about two different things here. Sorry for confusion, it looks like I wasn't clear enough. I was aiming that someone could state something like this `@Revision("NO_REVISION")` not knowing that "NO_REVISION" is reserved keyword for us. I can agree that it's not likely that someone will do it, but I think that we should be careful about it. If @smcvb agrees that it's a long shot, we can close this discussion. Otherwise, it would be cool to prevent it.

Reply:
It definitely is a minor thing, but I have to agree with @m1l4n54v1c. I'd rather be safe and ensure that your `RevisionSnapshortJury` works with any String in the `@Revision()` annotation than have somebody accidentally providing `NO_REVISION` in there which breaks the intent of your `RevisionSnapshortJury`.